### PR TITLE
Request creation is idempotent

### PR DESCRIPTION
**Pull request by ukstv**
Also present request in a uniform way

### CODE REVIEW

File: src/models/request-presentation.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import AnchorRepository from '../repositories/anchor-repository';
+import { InvalidRequestStatusError, RequestStatus } from './request-status';
+import awsCronParser from 'aws-cron-parser';
+import { config } from 'node-config-ts';
+import { Request } from './request';
+
+export class RequestPresentation {
```
Comment (stbrody) [inline src/models/request-presentation.ts:7]: class comment please

File: src/models/request-status.ts [hunk 2]
```diff
@@ -4,3 +4,9 @@ export enum RequestStatus {
   COMPLETED = 2,
   FAILED = 3,
 }
+
+export class InvalidRequestStatusError extends Error {
```
Comment (stbrody) [inline src/models/request-status.ts:8]: We have plans to introduce a proper error system based on https://github.com/ceramicnetwork/js-errors-utils.  Until we do that, I'd rather keep our errors simple and hierarchy free, so they're easy to update to the new system later.  Can you just use raw `Error` for now?

File: src/controllers/request-controller.ts [hunk 3]
```diff
@@ -117,46 +66,47 @@ export default class RequestController {
...
+        request.cid = cid.toString();
+        request.docId = docId;
+        request.status = RequestStatus.PENDING;
+        request.message = 'Request is pending.';
+
+        request = await this.requestRepository.createOrUpdate(request);
+
+        const cron = awsCronParser.parse(config.cronExpression);
+
+        return res.status(StatusCodes.CREATED).json({
```
Comment (stbrody) [inline src/controllers/request-controller.ts:96]: might as well move this case into requestPresentation and use it here, to keep everything in the same place.

**Review by stbrody — COMMENTED** [review]
a few small comments, but mostly looks good!

File: src/models/request-presentation.ts (hunk 1 above)
Comment (ukstv) [reply src/models/request-presentation.ts:7]: ok

File: src/models/request-status.ts (hunk 2 above)
Comment (ukstv) [reply src/models/request-status.ts:8]: While I agree on unification, I would like to leave it here as it is. Thanks to `never` argument it exhaustively checks if all the enum variants are handled.

Attached for reference: https://github.com/ceramicnetwork/js-errors-utils/issues/2

File: src/controllers/request-controller.ts (hunk 3 above)
Comment (ukstv) [reply src/controllers/request-controller.ts:96]: whoops, true

File: src/models/request-presentation.ts (hunk 1 above)
Comment (ukstv) [reply src/models/request-presentation.ts:10]: hopefully, can be resolved

File: src/models/request-status.ts (hunk 2 above)
Comment (stbrody) [reply src/models/request-status.ts:8]: wow, fascinating.  What if you removed the `default` case from the switch statement, and put the throw after the switch statement - wouldn't that have the same effect?

**Review by stbrody — APPROVED** [review]
I'd still prefer to remove the new Error class, but I'll leave the final decision to you.  Otherwise LGTM

File: src/models/request-status.ts (hunk 2 above)
Comment (ukstv) [reply src/models/request-status.ts:8]: Not at all. That's the point. With `throw` in `default` case, the compiler does not complain when you add a new variant to enum and forget to handle it.

Comment (stbrody) [reply src/models/request-status.ts:8]: If you have no default case, the compiler won't warn you if you don't handle an enum variant case?  That's pretty disappointing

**Review by oed — APPROVED** [review]
lgtm, mod merge requests.

**Comment by ukstv** [comment]
Ooh, that should have been squashed. Sorry for that.

**Pull request merged**